### PR TITLE
Support Traffic Splitting through `ServiceProfile.dstOverrides`

### PR DESCRIPTION
**Pull request by Pothulapati**
Initial part of #6054 

This PR 
- updates the destination logic to prevent the override of `ServiceProfiles.dstOverrides` when a `TS` is absent and no `dstOverrides` set. This means that any `serviceprofiles.dstOverrides` set by the user are correctly propagated to the proxy and thus making Traffic Splitting through service profiles work.
- This also updates the `profile_translator.toDstOverrides` to use the port from `GetProfiles` when there is no port in `dstOverrides.authority`

After these changes, The following `ServiceProfile` to perform
traffic splitting should work.

```yaml
apiVersion: linkerd.io/v1alpha2
kind: ServiceProfile
metadata:
  name: backend-svc.linkerd-trafficsplit-test-sp.svc.cluster.local
spec:
  dstOverrides:
  - authority: "backend-svc.linkerd-trafficsplit-test-sp.svc.cluster.local.:8080"
    weight: 500m
  - authority: "failing-svc.linkerd-trafficsplit-test-sp.svc.cluster.local.:8080"
    weight: 500m
```

This PR also adds an integration test i.e `TestTrafficSplitCliWithSP` which  checks for
Traffic Splitting through Service Profiles. This integration test follows the same pattern 
of other traffic split integration tests but Instead, we perform `linkerd viz stat` on the
client and check if there are expected server objects to be there as `linkerd viz stat ts`
does not _yet_ work with Service Profiles.

---
- update trafficsplit adapter to not do anything when ts are not present
- relax requirements of routes in sp-validator and Open API Validation
- nit: Update to the expectedLeaf to have the correct value
- add TestTrafficSplitCliWithSP integration test

### CODE REVIEW

File: controller/api/destination/traffic_split_adaptor.go [hunk 1]
```diff
@@ -65,14 +64,6 @@ func (tsa *trafficSplitAdaptor) publish() {
 			overrides = append(overrides, dst)
 		}
 		merged.Spec.DstOverrides = overrides
-	} else {
-		// If there is no traffic split, always return a destination override
```
Comment (adleong) [inline controller/api/destination/traffic_split_adaptor.go:69]: It looks like before this change, if there was no TrafficSplit, we would set the DstOverrides to be a single element list containing the dst.  After this change, if there is no TS and no DstOverrides are set by the user, it seems like we keep the DstOverrides as an empty list.  Is that change intentional?  Is it the behavior that we want?

Comment (Pothulapati) [reply controller/api/destination/traffic_split_adaptor.go:69]: Yep, That seems right. That is what I was expecting to do. Skip the overriding when there is no TS. This means that if any `dstOverrides` were set at SP, those are propogated as is. And as you said an empty list when both are not set.

I went with the following approach because the docs about `dstOverrides` mention that **if the list is empty, requests are sent to the destination as normal**.

```go
 	// If this list is non-empty, requests to this destination should instead be
	// split between the destinations in this list.  Each destination should
	// receive a portion of the requests proportional to its weight.  If this
	// list is empty, requests should be sent to this destination as normal.
	DstOverrides []*WeightedDst `protobuf:"bytes,3,rep,name=dst_overrides,json=dstOverrides,proto3" json:"dst_overrides,omitempty"`

```

is this not the behavior in practice?

Comment (adleong) [reply controller/api/destination/traffic_split_adaptor.go:69]: That sounds right, but since this is a change from how we do it today, I just wanted to confirm that you've tested it and that having an empty list of DstOverrides works as expected.

**Review by adleong — APPROVED** [review]

File: controller/api/destination/traffic_split_adaptor.go (hunk 1 above)
Comment (kleimkuhler) [reply controller/api/destination/traffic_split_adaptor.go:69]: @Pothulapati Do we know how this interacts with the proxy performing a `Get` request? The default override was added to close #4879. With this change, `GetProfile` for a service will not return destination overrides so I don't think we'll perform `Get` on the endpoints now. This could lead to the proxy not performing its load balancing on the endpoints.

File: test/integration/trafficsplit/testdata/applications-at-diff-ports.yaml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+kind: Service
+metadata:
+  name: slow-cooker
+spec:
+  selector:
+    app: slow-cooker
+  ports:
+  - name: metrics
+    port: 9999
+    targetPort: 9999
```
Comment (kleimkuhler) [inline test/integration/trafficsplit/testdata/applications-at-diff-ports.yaml:119]: Do we need a slow cooker service?

File: controller/api/destination/traffic_split_adaptor.go (hunk 1 above)
Comment (kleimkuhler) [reply controller/api/destination/traffic_split_adaptor.go:69]: So talking this through a little more we still do want to make sure `dst_overrides` for the destination profile is set if the service profile has none itself.

We should change the check here though to not be an `else` for `tsa.split != nil`. We instead would want a separate check for `if len(merged.Spec.DstOverrides) == 0`. If this is the case, we know that the traffic split added no overrides and the service profile has no overrides. Therefore, the response's `dst_overrides` should be set so that the proxy still performs its own load balancing.

Comment (Pothulapati) [reply controller/api/destination/traffic_split_adaptor.go:69]: Makes Sense. Updated!

File: test/integration/trafficsplit/testdata/applications-at-diff-ports.yaml (hunk 2 above)
Comment (Pothulapati) [reply test/integration/trafficsplit/testdata/applications-at-diff-ports.yaml]: Removed.

**Review by kleimkuhler — APPROVED** [review]
Cool looks good @Pothulapati.
